src/ring_theory/roots_of_unity.lean

```diff
@@ -590,6 +605,29 @@ begin
     exact hx }
 end
 
+lemma ne_zero' {n : ℕ+} (hζ : is_primitive_root ζ n) : ne_zero ((n : ℕ) : R) :=
```

Comment:
isn't this true in a bit more generality than an integral domain? i'll have a think...

Reply:
oh, the frobenius even requires primality! makes a lot of sense